full-stack-tests/presentation/src/learning-snippets/Hierarchies.test.ts

```diff
@@ -24,305 +28,1618 @@ describe("Learning Snippets", () => {
-      // respectively.The "B" rule has a label override of its own.
-      const ruleset: Ruleset = {
-        id: "example",
-        rules: [{
-          // This label override applies to all nodes in the hierarchy
-          ruleType: RuleTypes.LabelOverride,
-          label: `"Global: " & ThisNode.Label`,
+        const rootSubjectChildNodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset, parentKey: rootSubjectNodes[0].key });
+        expect(rootSubjectChildNodes).to.have.lengthOf(1).and.to.containSubset([{
+          key: { type: StandardNodeTypes.ECInstancesNode, instanceKeys: [{ className: "BisCore:RepositoryModel", id: "0x1" }] },
+        }]);
+
+        const repositoryModelChildNodes = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset, parentKey: rootSubjectChildNodes[0].key });
+        expect(repositoryModelChildNodes).to.have.lengthOf(11).and.to.containSubset([{
+          label: { displayValue: "DgnV8Bridge" },
+        }, {
+          label: { displayValue: "BisCore.RealityDataSources" },
+        }, {
+          label: { displayValue: "BisCore.DictionaryModel" },
+        }, {
+          label: { displayValue: "Properties_60InstancesWithUrl2.dgn" },
+        }, {
+          label: { displayValue: "DgnV8Bridge:D:\\Temp\\Properties_60InstancesWithUrl2.dgn, Default" },
+        }, {
+          label: { displayValue: "Converted Groups" },
+        }, {
+          label: { displayValue: "Converted Drawings" },
+        }, {
+          label: { displayValue: "Converted Sheets" },
+        }, {
+          label: { displayValue: "Definition Model For DgnV8Bridge:D:\\Temp\\Properties_60InstancesWithUrl2.dgn, Default" },
+        }, {
+          label: { displayValue: "Properties_60InstancesWithUrl2" },
         }, {
-          ruleType: RuleTypes.RootNodes,
-          specifications: [{
-            specType: ChildNodeSpecificationTypes.CustomNode,
-            type: "A",
-            label: "A",
+          label: { displayValue: "Properties_60InstancesWithUrl2" },
+        }]);
+
+        const repositoryModelNodes2 = await Presentation.presentation.getNodes({ imodel, rulesetOrId: ruleset, parentKey: repositoryModelChildNodes.find((n) => n.label.displayValue === "DgnV8Bridge")!.key });
+        expect(repositoryModelNodes2).to.have.lengthOf(1).and.to.containSubset([{
+          key: { type: StandardNodeTypes.ECInstancesNode, instanceKeys: [{ className: "BisCore:RepositoryModel", id: "0x1" }] },
+        }]);
+      });
+
+      it("uses `priority` attribute", async () => {
+        // __PUBLISH_EXTRACT_START__ Hierarchies.Specification.Priority.Ruleset
+        // The ruleset has two specifications that return nodes for `bis.PhysicalModel` and `bis.SpatialCategory`
+        // respectively. The specifications have different priorities and higher priority rule is handled
+        // first - that's the reason the `bis.SpatialCategory` node appears first in the result.
```

Comment:
```suggestion
        // This ruleset produces a list of `bis.PhysicalModel` and `bis.SpatialCategory` instances and groups them by
        // class. "Spatial Category" group will appear first because it has been given a higher priority value.
```